### PR TITLE
[RA2 Appendix]: Describe best practices for workload isolation with Kubernetes : initial layout with notes

### PR DESCRIPTION
**Pull request by sb1975**
This is to address the Multitenancy for CNF topic discussed during the opendev meeting - https://github.com/cntt-n/CNTT/issues/1878

### CODE REVIEW

File: doc/ref_arch/kubernetes/chapters/appendix-b.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+* [B.1 Overview](#B.1)
+* [B.2 Multitenancy Models for CNFs](#B.2)
+* [B.3 Solution Areas](#B.3)
+
+
+<a name="B.1"></a>
+## B.1 Overview
+
+Problem statement: A single Kubernetes cluster does not provide hard multitenancy by design. Within a Cluster, Namespace is a mechanism to provide Soft isolation multitenancy.
```
Comment (CsatariGergely) [inline doc/ref_arch/kubernetes/chapters/appendix-b.md:15]:
```suggestion
Problem statement: A single Kubernetes cluster does not provide hard multitenancy by design. Within a Cluster, Kubernetes Namespace is a mechanism to provide Soft isolation multitenancy.
```

We should somehow differentiate between kernel namespaces and Kubernetes namespaces.

File: doc/ref_arch/kubernetes/chapters/appendix-b.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+Namespace does provide isolation by means of RBAC, Resource Isolation and Network Policy, however they are still within the same trust domain and a potential breach of Cluster Admin Role could lead to the Blast Radius across the entire Cluster and all its namespaces.
```
Comment (CsatariGergely) [inline doc/ref_arch/kubernetes/chapters/appendix-b.md:16]:
```suggestion
A Kubernetes Namespace does provide isolation by means of role based access control (RBAC), Resource Isolation and Network Policy, however they are still within the same trust domain and a potential breach of Cluster Admin Role could lead to the Blast Radius across the entire Cluster and all its Kubernetes Namespaces.
```

File: doc/ref_arch/kubernetes/chapters/appendix-b.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+So there is a need to define various use cases or ways to build Multitenancy Deployment Models and define the Best Practices to secure each Model.
+
+<p align="left"><img src="../figures/Model2-cluster.png" alt="scope" title="Scope" width="50%"/></p>
```
Comment (CsatariGergely) [inline doc/ref_arch/kubernetes/chapters/appendix-b.md:19]: Please add an editable version to doc/ref_arch/kubernetes/artefacts/

File: doc/ref_arch/kubernetes/chapters/appendix-b.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+<a name="B.1"></a>
+## B.1 Overview
+
+Problem statement: A single Kubernetes cluster does not provide hard multitenancy by design. Within a Cluster, Namespace is a mechanism to provide Soft isolation multitenancy.
+Namespace does provide isolation by means of RBAC, Resource Isolation and Network Policy, however they are still within the same trust domain and a potential breach of Cluster Admin Role could lead to the Blast Radius across the entire Cluster and all its namespaces.
+So there is a need to define various use cases or ways to build Multitenancy Deployment Models and define the Best Practices to secure each Model.
+
+<p align="left"><img src="../figures/Model2-cluster.png" alt="scope" title="Scope" width="50%"/></p> 
+<p align="left"><img src="../figures/Model2-ns.png" alt="scope" title="Scope" width="50%"/></p>
```
Comment (CsatariGergely) [inline doc/ref_arch/kubernetes/chapters/appendix-b.md:20]: Please add an editable version to doc/ref_arch/kubernetes/artefacts/

File: doc/ref_arch/kubernetes/chapters/appendix-b.md (hunk 3 above)
Comment (CsatariGergely) [reply doc/ref_arch/kubernetes/chapters/appendix-b.md:19]: * Indicate the different vendors in the figure

File: doc/ref_arch/kubernetes/chapters/appendix-b.md (hunk 4 above)
Comment (CsatariGergely) [reply doc/ref_arch/kubernetes/chapters/appendix-b.md:20]: - Indicate different vendors and that these namespaces can be in one cluster

File: doc/ref_arch/kubernetes/chapters/appendix-b.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+Use cases:
+1 Two CNF-s which are in the same trust domain (e.g.: they are from the same vendor) are running in a container infrastructure
+2 Two CNF-s which are in different trust domains (e.g.: they are from different vendors) are running in a container infrastructure
+
+
+<a name="B.2"></a>
+## B.2 Multitenancy Models for CNFs
+Solution Models :
+1. Seperate Namespace per Vendor within Single Kubernetes Cluster
```
Comment (CsatariGergely) [inline doc/ref_arch/kubernetes/chapters/appendix-b.md:29]:
```suggestion
1. Seperate Kubernetes Namespace per Vendor within Single Kubernetes Cluster
```

File: doc/ref_arch/kubernetes/chapters/appendix-b.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+* [B.1 Overview](#B.1)
+* [B.2 Multitenancy Models for CNFs](#B.2)
+* [B.3 Solution Areas](#B.3)
+
+
+<a name="B.1"></a>
+## B.1 Overview
+
+Problem statement: A single Kubernetes cluster does not provide hard multitenancy by design. Within a Cluster, Kubernetes Namespace is a mechanism to provide Soft isolation multitenancy.
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/appendix-b.md:15]: do you need to define "hard" and "soft" isolation?

File: doc/ref_arch/kubernetes/chapters/appendix-b.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
+[<< Back](../../kubernetes)
+# Appendix B - Guidance For workload isolation(Multitenancy) with Kubernetes for application Vendors
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/appendix-b.md:2]: nit
```suggestion
# Appendix B - Guidance For workload isolation (Multitenancy) with Kubernetes for application Vendors
```

File: doc/ref_arch/kubernetes/chapters/appendix-b.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+<a name="B.3"></a>
+## B.3 Solution Areas
+The scope is to identify the solution area which is needed to secure the CNF workloads. Securing the platform might happen as part of it but not directly the focus or objective here.
+1. Isolation of Platform and Tenant Deployment based on the Solution Model
+2. Seperate CICD, Manifest and Image Repository for Platform and Tenants
+3. Isolation of Networking for CNF Workloads
+4. Securing the CNF Workload Network Traffic using Network Policy and Service Mesh
+5. Seperate Storage Backend for each CNF Workloads
+6. RBAC and secrets Management for CNF Workload
+7. Seperate Isolated view of Logging,Monitoring,Alerting and Tracing for CNF Workloads
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/appendix-b.md:41]:
```suggestion
7. Separate Isolated view of Logging, Monitoring, Alerting and Tracing for CNF Workloads
```

**Comment by pgoyal01** [comment]
multiple spelling typo's: "seperate" instead of "separate"

File: doc/ref_arch/kubernetes/chapters/appendix-b.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+<a name="B.2"></a>
+## B.2 Multitenancy Models for CNFs
+Solution Models :
+1. Seperate Kubernetes Namespace per Vendor within Single Kubernetes Cluster
+2. Seperate Kubernetes Cluster per Vendor
+
+<a name="B.3"></a>
+## B.3 Solution Areas
+The scope is to identify the solution area which is needed to secure the CNF workloads. Securing the platform might happen as part of it but not directly the focus or objective here.
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/appendix-b.md:34]: what about separate networking?

File: doc/ref_arch/kubernetes/chapters/appendix-b.md [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+
+<a name="B.1"></a>
+## B.1 Overview
+
+Problem statement: A single Kubernetes cluster does not provide hard multitenancy by design. Within a Cluster, Kubernetes Namespace is a mechanism to provide Soft isolation multitenancy.
+A Kubernetes Namespace does provide isolation by means of role based access control (RBAC), Resource Isolation and Network Policy, however they are still within the same trust domain and a potential breach of Cluster Admin Role could lead to the Blast Radius across the entire Cluster and all its Kubernetes Namespaces.
+So there is a need to define various use cases or ways to build Multitenancy Deployment Models and define the Best Practices to secure each Model.
+
+<p align="left"><img src="../figures/Model2-cluster-isolation.png" alt="scope" title="Scope" width="50%"/></p>
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/appendix-b.md:19]: In the diagram headings, can there be a space between "Model" and the number (1 or 2)?

What is the "bus" in both diagrams?

**Review by tomkivlin — APPROVED** [review]

**Review by sukhdevkapur — APPROVED** [review]

**Review by petorre — APPROVED** [review]

**Comment by CsatariGergely** [comment]
Looks okay, but please add the figures in editable format to a `ref_arch/kubernetes\artefacts` directory.

**Review by scottsteinbrueck — APPROVED** [review]

**Pull request merged**